app/controllers/api/v1/packages_controller.rb

```diff
@@ -36,8 +36,12 @@ class PackagesController < Api::V1::ApiController
 
       api :GET, "/v1/packages", "get all packages for the item"
       def index
-        @packages = @packages.find(params[:ids].split(",")) if params[:ids].present?
-        render json: @packages, each_serializer: serializer, include_orders_packages: true
+        records = @packages #security
+        records = records.filter("state" => params["state"]) if params["state"].present?
+        records = records.search({search_text: params["searchText"]}) if params["searchText"]
```

Comment:
do we have specs for those ?

Reply:
I am working on specs right now. Will add those in a while.

Reply:
Cool, remember that as much as we can, specs should be written _before_ the implementation : )